src/Broadway/EventStore/MongoDBEventStore.php

```diff
@@ -0,0 +1,371 @@
+<?php
+
+/*
+ * This file is part of the broadway/broadway package.
+ *
+ * (c) Qandidate.com <opensource@qandidate.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Broadway\EventStore;
+
+use Broadway\Domain\DateTime;
+use Broadway\Domain\DomainEventStream;
+use Broadway\Domain\DomainEventStreamInterface;
+use Broadway\Domain\DomainMessage;
+use Broadway\Domain\Metadata;
+use Broadway\EventStore\Exception\InvalidIdentifierException;
+use Broadway\Serializer\SerializerInterface;
+use Broadway\UuidGenerator\UuidGeneratorInterface;
+use Doctrine\MongoDB\Collection;
+use Doctrine\MongoDB\Database;
```

Comment:
Is there any value in using the Doctrine classes instead of the raw driver? This class seems comparable to the session and cache adapters I worked on for Symfony and Zend, since it consists of just a few direct read and write ops.

As noted below, Doctrine doesn't yet support the MongoWriteBatch classes (for batch ops via the write commands in 2.6+), nor the `createIndex()` method (although the deprecated `ensureIndex()` method is an alias). Were you to make the switch, I'd suggest requiring at least `>=1.5.0` of the `mongo` extension (although it wouldn't hurt to bump to `1.6.0` given that this is a new integration).